coral/src/app/features/topics/request/schemas/topic-request-form.ts

```diff
@@ -0,0 +1,198 @@
+
+const topicpartitionsField = z.string();
+
+const replicationfactorField = z.string();
+
+const environmentField = z.object({
+  name: z.string(),
+  id: z.string(),
+  maxReplicationFactor: z.number().optional(),
+  maxPartitions: z.number().optional(),
+  defaultPartitions: z.number().optional(),
+  defaultReplicationFactor: z.number().optional(),
+  topicNamePrefix: z.string().optional(),
+  topicNameSuffix: z.string().optional(),
+});
+
+const formSchema = z
+  .object({
+    environment: environmentField,
+    topicpartitions: topicpartitionsField,
+    replicationfactor: replicationfactorField,
+    topicname: topicNameField,
+    remarks: z.string(),
+    description: z.string(),
+  })
+  .superRefine(validateTopicPartitions)
+  .superRefine(validateReplicationFactor)
+  .superRefine(validateTopicName);
+
+function validateReplicationFactor(
+  val: {
+    environment: z.infer<typeof environmentField>;
+    replicationfactor: z.infer<typeof replicationfactorField>;
+  },
+  ctx: RefinementCtx
+) {
+  const {
+    environment: { maxReplicationFactor },
+    replicationfactor,
+  } = val;
+  if (
+    isNumber(maxReplicationFactor) &&
+    parseInt(replicationfactor, 10) > maxReplicationFactor
+  ) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.too_big,
+      inclusive: true,
+      maximum: maxReplicationFactor,
+      type: "number",
+      path: ["replicationfactor"],
+      message: `${replicationfactor} can not be bigger than ${maxReplicationFactor}`,
+    });
+  }
+}
+
+function validateTopicPartitions(
+  val: {
+    environment: z.infer<typeof environmentField>;
+    topicpartitions: z.infer<typeof topicpartitionsField>;
+  },
+  ctx: RefinementCtx
+) {
+  const {
+    environment: { maxPartitions },
+    topicpartitions,
+  } = val;
+  if (
+    isNumber(maxPartitions) &&
+    parseInt(topicpartitions, 10) > maxPartitions
+  ) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.too_big,
+      inclusive: true,
+      maximum: maxPartitions,
+      type: "number",
+      path: ["topicpartitions"],
+      message: `${topicpartitions} can not be bigger than ${maxPartitions}`,
+    });
+  }
+}
+
+function validateTopicName(
+  val: {
+    environment: z.infer<typeof environmentField>;
+    topicname: z.infer<typeof topicNameField>;
+  },
+  ctx: RefinementCtx
+) {
+  const {
+    environment: { topicNamePrefix, topicNameSuffix },
+    topicname,
+  } = val;
+  if (topicNamePrefix !== undefined && !topicname.startsWith(topicNamePrefix)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      fatal: true,
+      message: `Topic name must start with "${topicNamePrefix}".`,
+      path: ["topicname"],
+    });
+  }
+  if (topicNameSuffix !== undefined && !topicname.endsWith(topicNameSuffix)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      fatal: true,
+      message: `Topic name must end with "${topicNameSuffix}".`,
+      path: ["topicname"],
+    });
+  }
+}
+
+const useExtendedFormValidationAndTriggers = (
```

Comment:
Maybe a more detailed comment about why this is needed and how we achieve the desired result would be nice. I remember why _now_, but I don't know if it will be as crystal clear in a few months haha!